### PR TITLE
Switch Geocoder to use Mapbox API instead of Google Maps API

### PR DESCRIPTION
**Pull request by daynew**
Soon we will no longer be able to use the Google Maps API. This change switches our `Geocoder.search` method calls to use the Mapbox API instead.

## Links
- [jira](https://codedotorg.atlassian.net/browse/FND-1300)

## Testing story
* Manually tested in the dashboard console.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by bethanyaconnor — APPROVED** [review]
Looks good to me!

File: lib/cdo/geocoder.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def self.find_potential_street_address(text)
     results = Geocoder.search(first_number_to_end)
     return nil if results.empty?
 
-    if results.first.types.include?('street_address')
+    if results.first.address
```
Comment (bethanyaconnor) [inline lib/cdo/geocoder.rb]: Putting a comment here but I was trying to run this and it was crashing in `summarize` as a number of those attributes , such as `street_address`, don't exist for Mapbox. Can you take a look?

**Comment by bethanyaconnor** [comment]
I'm a bit more worried about this change than I was yesterday. How much have you tested? I'm mostly concerned by the fact that the responses from the two APIs seem to be quite different and a lot of places in our code base use this class

**Comment by daynew** [comment]
Added tests and will manually test in staging. I will get a proper review during the week.

**Pull request merged**